src/utilities/get-directory-contents.ts

```diff
@@ -25,44 +26,43 @@ export async function getDirectoryContents<T extends object>(
 
         let itemName: string;
         let fileName: string;
-        let fullPath: string;
+        let url = pathToFileURL(fullPathStr);
         if (isDir) {
-          itemName = fullPathStr.split('/').at(-1)!;
+          itemName = url.toString().split('/').at(-1) ?? '';
           fileName = 'index.js';
-          fullPath = path.join(fullPathStr, 'index.js');
+          url = pathToFileURL(path.join(fullPathStr, fileName));
```

Comment:
If we are only ever using the string version of `url`, I think we should just `toString()` it here instead of calling `toString()` twice in the `for` loop below (lines 51 and 58)

Reply:
Anddddd done. I also removed a bit of code that didn't needed to be returned from the function at all.